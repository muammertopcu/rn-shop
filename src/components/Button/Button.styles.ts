import styled from 'styled-components/native';
import {Text, TouchableOpacity} from 'react-native';
import {moderateScale} from 'react-native-size-matters';

export const Container = styled(TouchableOpacity)`
  background-color: #000;
  padding: 10px;
  border-radius: 5px;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  gap: ${moderateScale(10)}px;
`;

export const Label = styled(Text)`
  color: #fff;
  font-size: ${moderateScale(16)}px;
`;
