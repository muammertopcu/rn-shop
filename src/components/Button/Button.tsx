import React from 'react';
import {Icon} from '@components';
import {Container, Label} from './Button.styles.ts';
import type {ButtonProps} from './Button.types.ts';

const Button = ({label, onPress, icon, ...props}: ButtonProps) => {
  return (
    <Container testID={'button-container'} onPress={onPress} {...props}>
      {icon && <Icon name={icon} />}
      <Label>{label}</Label>
    </Container>
  );
};

export default Button;
