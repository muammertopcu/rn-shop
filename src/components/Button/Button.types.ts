import {TouchableOpacityProps} from 'react-native';
import {icons} from '@constants';

export interface ButtonProps extends TouchableOpacityProps {
  label: string;
  onPress?: () => void;
  icon?: keyof typeof icons;
}
