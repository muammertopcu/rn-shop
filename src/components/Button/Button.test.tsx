import React from 'react';
import Button from './Button.tsx';
import {renderWithProviders} from '@utils';

describe('<Button />', () => {
  it('should render correctly', () => {
    const {getByTestId, toJSON} = renderWithProviders(<Button label={'Test'} />);

    expect(getByTestId('button-container')).toBeDefined();
    expect(toJSON()).toMatchSnapshot();
  });
});
