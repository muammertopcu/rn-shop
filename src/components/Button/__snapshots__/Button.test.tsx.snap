// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "borderBottomLeftRadius": 5,
      "borderBottomRightRadius": 5,
      "borderTopLeftRadius": 5,
      "borderTopRightRadius": 5,
      "flexDirection": "row",
      "gap": 15.714285714285714,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
  testID="button-container"
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 25.142857142857142,
      }
    }
  >
    Test
  </Text>
</View>
`;
